protostar/self/conftest.py

```diff
@@ -0,0 +1,12 @@
+from .protostar_compatibility_with_project_checker import (
+    ProtostarVersionProviderProtocol,
+    parse_protostar_version,
+)
+
+
+class ProtostarVersionProviderDouble(ProtostarVersionProviderProtocol):
```

Comment:
One may say that this is a `Fake` not a `Stub`. I'd rather stick to the convention `TestDouble` (I'll change the suffix from Double to TestDouble). 

> Why not use a mock? Would be much simpler

`MockerFixture` needs to be used per test case, which makes the test less readable. Also, mocked values are slightly harder to maintain because refactoring breaks such tests.

Reply:
I think `Double` is just a bad word because has vague meaning, If you prefer stick to `Fake` it is also fine

Reply:
> I think Double is just a bad word because has vague meaning

That's why I wanted to rename it `TestDouble`.  Stubs, Fakes, Mocks are often misused.

Renamed `TestDouble`s in [08a6b28](https://github.com/software-mansion/protostar/pull/897/commits/08a6b28366385040c2daaa114fac5dd235fabd23)